src/main/java/uk/gov/hmcts/reform/ccd/async/TimedJobExecutor.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.hmcts.reform.ccd.async;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+@Slf4j
+@Component
+public class TimedJobExecutor {
+    private final ExecutorService executor = Executors.newVirtualThreadPerTaskExecutor();
```

Comment:
Why this needs to be private field. If we are shutting down this in runWithTimeout method then we can create this locally in runWithTimeout , use it and then close it. 
ExecutorService executor = Executors.newVirtualThreadPerTaskExecutor();